File: tuan2/PhanA/bai31.jl
using Printf

function main()
    tokens = split(read(stdin, String))
    if isempty(tokens)
        return
    end

    n = parse(Int, tokens[1])
    first_val = parse(Int, tokens[2])

    sum_val = first_val
    min_val = first_val
    max_val = first_val

    for i in 2:(n + 1)
        x = parse(Int, tokens[i])
        sum_val += x
        if x < min_val
            min_val = x
        end
        if x > max_val
            max_val = x
        end
    end

    avg = sum_val / n
    @printf("%d %.4f %d %d\n", sum_val, avg, min_val, max_val)
end

main()

File: tuan2/PhanA/bai31.py
import sys

def main():
    # sys.stdin.read().split() là phương thức chuẩn đọc nhanh dữ liệu
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    n = int(input_data[0])
    first_val = int(input_data[1])
    
    total_sum = first_val
    min_val = first_val
    max_val = first_val
    
    for i in range(2, n + 1):
        x = int(input_data[i])
        total_sum += x
        if x < min_val:
            min_val = x
        if x > max_val:
            max_val = x
            
    avg = total_sum / n
    print(f"{total_sum} {avg:.4f} {min_val} {max_val}")

if __name__ == "__main__":
    main()